packages/frontend/src/components/ProjectAccess/ProjectAccess.tsx

```diff
@@ -126,6 +129,67 @@ const ProjectAccess: FC<ProjectAccessProps> = ({
             projectUuid,
         }),
     );
+    const inheritedRoles = useMemo(() => {
+        if (!organizationUsers) return {};
+        return organizationUsers.reduce<Record<string, InheritedRoles>>(
+            (acc, orgUser) => {
+                return {
+                    ...acc,
+                    [orgUser.userUuid]: [
+                        {
+                            type: 'organization',
+                            role: convertOrganizationRoleToProjectRole(
+                                orgRoles[orgUser.userUuid],
+                            ),
+                        },
+                        {
+                            type: 'group',
+                            role: groupRoles[orgUser.userUuid],
+                        },
+                        {
+                            type: 'project',
+                            role: projectRoles[orgUser.userUuid],
+                        },
+                    ],
+                };
+            },
+            {},
+        );
+    }, [organizationUsers, orgRoles, groupRoles, projectRoles]);
+
+    const usersWithProjectRole = useMemo(() => {
+        if (!organizationUsers) return [];
+
+        return organizationUsers.map((orgUser) => {
+            const highestRole = getHighestProjectRole(
+                inheritedRoles[orgUser.userUuid],
+            );
+            const hasProjectRole = !!projectRoles[orgUser.userUuid];
+            const inheritedRole = highestRole?.role
+                ? convertProjectRoleToOrganizationRole(highestRole.role)
+                : orgUser.role;
+            return {
+                ...orgUser,
+                finalRole: hasProjectRole
```

Comment:
I don't want to overwrite the `org role` as it might be used in nested componets (at the moment we don't use directly `role` on ProjectAccessRow, but we might use it in the future. We only wan this new field for filtering, see Fuse method below